### PR TITLE
feat: add ci for nightly release

### PR DESCRIPTION
**Pull request by usamoi**
Nightly releases will be built on 00:00 UTC+8 if there are any commits in the past 24 hours.

### CODE REVIEW

**Review by gaocegege — DISMISSED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          touch ./target/vectors_0.0-0_amd64/DEBIAN/postinst
+          chmod +x ./target/vectors_0.0-0_amd64/DEBIAN/postinst
+          echo "Package: vectors" >> ./target/vectors_0.0-0_amd64/DEBIAN/control
+          echo "Version: 0.0" >> ./target/vectors_0.0-0_amd64/DEBIAN/control
+          echo "Architecture: amd64" >> ./target/vectors_0.0-0_amd64/DEBIAN/control
+          echo "Maintainer: example@example.com" >> ./target/vectors_0.0-0_amd64/DEBIAN/control
+          echo "Description: No descrption." >> ./target/vectors_0.0-0_amd64/DEBIAN/control
+          echo "su postgres -c \"psql -U postgres -c 'ALTER SYSTEM SET shared_preload_libraries = \"vectors\"'\"" >> ./target/vectors_0.0-0_amd64/DEBIAN/postinst
+          mv ./target/release/vectors-pg$(psql --version | awk -F"[ ,.]" '{print $3}')/usr ./target/vectors_0.0-0_amd64/usr
+          dpkg-deb --build --root-owner-group ./target/vectors_0.0-0_amd64
```
Comment (VoVAllen) [inline .github/workflows/release.yml:69]: We should support dynamic version

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        run: |
+          cargo pgrx package
+          # deb
+          mkdir ./target/vectors_0.0-0_amd64
+          mkdir ./target/vectors_0.0-0_amd64/DEBIAN
+          touch ./target/vectors_0.0-0_amd64/DEBIAN/control
+          touch ./target/vectors_0.0-0_amd64/DEBIAN/postinst
+          chmod +x ./target/vectors_0.0-0_amd64/DEBIAN/postinst
+          echo "Package: vectors" >> ./target/vectors_0.0-0_amd64/DEBIAN/control
+          echo "Version: 0.0" >> ./target/vectors_0.0-0_amd64/DEBIAN/control
```
Comment (VoVAllen) [inline .github/workflows/release.yml:63]: ditto

**Review by VoVAllen — CHANGES_REQUESTED** [review]
Please change the trigger to the new tags, and use tag's version as the extension's version.

Example (https://github.com/tensorchord/envd/blob/main/.github/workflows/release.yml):
```yaml
on:
  push:
    tags:
      - 'v*'
``` 
and use `${{ github.event.release.tag_name }}` to get the version


Also better to change the building process to docker instead of pure github action. This makes local debug and build much easier

**Comment by usamoi** [comment]
@VoVAllen I'm not sure when will the version move from 0.0.0...

**Comment by VoVAllen** [comment]
@usamoi I already made one https://github.com/tensorchord/pgvecto.rs/releases/tag/v0.1.0. However I think it should be 0.0.1 if we're not confident now. We can start with 0.0.1, it's fine. What do you think?

**Comment by usamoi** [comment]
r?

**Review by gaocegege — COMMENTED** [review]
@VoVAllen

**Review by VoVAllen — APPROVED** [review]
Let's merge this first to see whether it works

**Pull request merged**